File: Entrada de Dados/AlfaNum/alfanum.scala
import scala.io.StdIn.readLine
import scala.util.control.Breaks._

object Main {
  def VALIDA_HEXA(VALOR: String): Boolean = {
    for (c <- VALOR) {
      val CARACTERE = c.toUpper
      if (!(Character.isDigit(CARACTERE) || (CARACTERE >= 'A' && CARACTERE <= 'F')))
        return false
    }
    true
  }

  def main(args: Array[String]): Unit = {
    println()
    var VALOR_HEX = ""
    var sairLoop = false
    breakable {
      do {
        print("Entre valor hexadecimal de 4 caracteres: ")
        VALOR_HEX = readLine()
        if (VALOR_HEX.length != 4) {
          println("Valor invalido, deve ter exatamente 4 caracteres.\n")
        } else if (!VALIDA_HEXA(VALOR_HEX)) {
          println("Valor invalido, apenas caracteres [0-9] e [A-F].\n")
        } else {
          sairLoop = true
          break
        }
      } while (!sairLoop)
    }
    VALOR_HEX = VALOR_HEX.toUpperCase()
    println("Valor fornecido: " + VALOR_HEX + "\n")
  }
}

File: Entrada de Dados/AlfaNum/alfanum.py
def VALIDA_HEXA(VALOR):
  for CARACTERE in VALOR:
    if not (CARACTERE.isdigit() or (CARACTERE >= 'A' and CARACTERE <= 'F') or (CARACTERE >= 'a' and CARACTERE <= 'f')):
      return False
  return True

print("")
VALOR_HEX = ""
SAIR_LOOP = False
while not SAIR_LOOP:
  VALOR_HEX = input("Entre valor hexadecimal de 4 caracteres: ").strip()
  if (len(VALOR_HEX) != 4):
    print("Valor invalido, deve ter exatamente 4 caracteres.")
    print()
    continue
  if (not VALIDA_HEXA(VALOR_HEX)):
    print("Valor invalido, apenas caracteres [0-9] e [A-F].")
    print()
    continue
  SAIR_LOOP = True
VALOR_HEX = VALOR_HEX.upper()
print("Valor fornecido: " + VALOR_HEX)
print()
